mmv1/templates/terraform/pre_delete/firebase_database_instance.go.erb

```diff
@@ -0,0 +1,26 @@
+<%- # the license inside this block applies to this file
+	# Copyright 2022 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+
+// start of customized code
+if d.Get("state").(string) == "ACTIVE" {
+  if err := disableRTDB(config, d, project, billingProject, userAgent); err != nil {
+    return err
+  }
+}
+if d.Get("type").(string) == "DEFAULT_DATABASE" {
+	log.Printf("[DEBUG] Default Firebase Database Instance %q cannot be deleted, left disabled", d.Id())
+	return nil
+}
```

Comment:
The tests you have are great, but I wasn't able to figure out if this line was under test. I suppose `firebase_database_instance_default.tf.erb` covers this during the destroy step?

And to clarify the outcome, this means that the database will be removed from state (thus matching the user's config), and appear to have been deleted, with the DEBUG line indicating the actual state of disabled on the server. Is that correct? If so, I don't know if there is precedence for this pattern, but I believe we would want something more visible like a [WARN], since hardly any users will be running Terraform with debug logging.

Reply:
Perfect, sounds good